File: source-py3/guessing_game.py
#! python3
import random

class GuessingName:
    
    def __init__(self):
        self.__secret_number = random.randint(1, 101)

    def run(self):
        print("Guess the number!")

        guess = 0
        while True:
            guess = input("Please input your guess.")

            guess = int(guess)

            if guess == self.__secret_number:
                print('You win!')
                break
            elif guess < self.__secret_number:
                print('Too small!')
            else:
                print('Too big!')

GuessingName().run()

File: source-lang/go/playground/guessing_game.go
package main

import (
	"fmt"
	"math/rand"
)

type guessingGame struct {
	secretNumber int
}

func newGuessingGame() guessingGame {
	return guessingGame{secretNumber: rand.Int() % 100}
}

func (i guessingGame) run() {
	fmt.Println("Guess the number!")

	var guess int
	for {
		fmt.Println("Please input your guess.")

		fmt.Scanln(&guess)

		if guess == i.secretNumber {
			fmt.Println("You win!")
			break
		} else if guess < i.secretNumber {
			fmt.Println("Too small!")
		} else {
			fmt.Println("Too big!")
		}
	}
}

func main() {
	fmt.Println("Hello, world!")
	game := newGuessingGame()
	game.run()
}
